django/contrib/admin/templatetags/admin_list.py

```diff
@@ -181,6 +182,13 @@ def _boolean_icon(field_val):
     return format_html('<img src="{}" alt="{}" />', icon_url, field_val)
 
 
+def _coerce_field_name(field_name):
+    """
+    Coerce a field_name (which may be a callable) to a string.
+    """
+    return field_name.__name__ if callable(field_name) else field_name
```

Comment:
Three suggestions:

1. Strip `<>` chars and print "lambda".
2. Fallback to the original suggestion proposed in the ticket: Use `callable.short_description` if available.
3. Wrap `for field_name in cl.list_display:` with `enumerate()` and use the index in `_coerce_field_name`. (e.g. field-lambda-1_year)

The third suggestion is probably overkill for that case.